internal/api_test.go

```diff
@@ -250,43 +251,79 @@ func TestDelayedLogFlushing(t *testing.T) {
 	f, c, cleanup := setup()
 	defer cleanup()
 
-	http.HandleFunc("/quick_log", func(w http.ResponseWriter, r *http.Request) {
+	http.HandleFunc("/slow_log", func(w http.ResponseWriter, r *http.Request) {
 		logC := WithContext(netcontext.Background(), r)
 		fromContext(logC).apiURL = c.apiURL // Otherwise it will try to use the default URL.
 		Logf(logC, 1, "It's a lovely day.")
 		w.WriteHeader(200)
+		time.Sleep(1200 * time.Millisecond)
 		w.Write(make([]byte, 100<<10)) // write 100 KB to force HTTP flush
 	})
 
 	r := &http.Request{
 		Method: "GET",
 		URL: &url.URL{
 			Scheme: "http",
-			Path:   "/quick_log",
+			Path:   "/slow_log",
 		},
 		Header: c.req.Header,
 		Body:   ioutil.NopCloser(bytes.NewReader(nil)),
 	}
 	w := httptest.NewRecorder()
 
-	// Check that log flushing does not hold up the HTTP response.
-	start := time.Now()
-	handleHTTP(w, r)
-	if d := time.Since(start); d > 10*time.Millisecond {
-		t.Errorf("handleHTTP took %v, want under 10ms", d)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		handleHTTP(w, r)
+	}()
+	// Check that the log flush eventually comes in.
+	time.Sleep(1200 * time.Millisecond)
```

Comment:
Was this meant to be 1500 ms? Comment below says 1.5s.

Reply:
The comment was wrong; fixed